ui/src/app/lib/atlasmap-data-mapper/components/document.definition.component.ts

```diff
@@ -71,9 +71,9 @@ import { ModalWindowComponent } from './modal.window.component';
                             <document-field-detail #fieldDetail *ngFor="let f of docDef.fields" [modalWindow]="modalWindow"
                                 [field]="f" [cfg]="cfg" [lineMachine]="lineMachine"></document-field-detail>
                             <div class="FieldDetail"
-                                *ngIf="!searchMode && docDef.initCfg.type.isPropertyOrConstant() && !docDef.fields.length">
+                                *ngIf="!searchMode && docDef.isPropertyOrConstant() && !docDef.fields.length">
```

Comment:
Same as my previous comment, you do not want to attach functions execution to an structural directive (such as `NgIf` here) since those will be executed on each each pass of the change detector.

Fix: move `docDef.isPropertyOrConstant()` execution to an assignation expression inside `OnInit` in the component controller that binds its value to a `isPropertyOrConstant: boolean` class field and then bind that field inside the `*ngIf` conditional expression.

Regarding `!docDef.fields.length`, that will raise an exception if `docDef.fields` does not exist. I do not know the  internals of its implementation but bear that in mind. Perhaps a more type-safety way to check the whole conditional would be (following up with my previous comment above):

`*ngIf="!searchMode && docDef.isPropertyOrConstant && (!docDef.fields || !docDef.fields.length)"`

Reply:
This required to hide the `type` property behind the setter/getter, to capture `isPropertyOrConstant` when `type` is set.